python/cudf/tests/test_dataframe_copy.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) 2018, NVIDIA CORPORATION.
+
+import pytest
+
+import numpy as np
+import pandas as pd
+
+from cudf.dataframe.dataframe import DataFrame
+
+"""
+DataFrame copy expectations
+* A shallow copy constructs a new compound object and then (to the extent
+  possible) inserts references into it to the objects found in the original.
+* A deep copy constructs a new compound object and then, recursively, inserts
+  copies into it of the objects found in the original.
+
+  A cuDF DataFrame is a compound object containing a few members:
+  _index, _size, _cols,
+  and _iter_count, _iter_keys # temporary?
+  where _cols is an OrderedDict and _iter_keys is a list
+"""
+
+
+def test_pandas_dataframe_copy():
```

Comment:
We shouldn't be running tests for only Pandas.

Reply:
I agree. I included it because it tests and demonstrates us mirroring Pandas. In theory if Pandas introduced a change, we'd catch it. Maybe we don't care that much, if we do?

Reply:
What I'd propose is having a test where we compare the results of running pandas.copy and cudf.copy and ensure the behaviors match. Three variations: default parameter, deep=False, and deep=True. That way we test that shallow + deep copies work as expected for cudf and ensure our behavior is consistent with Pandas's as the versions change.

Reply:
Agreed, I'm updating this up now.